controllers/articlesController.js

```diff
@@ -93,4 +82,154 @@ export default class PostController {
       });
     }
   }
+
+  /**
+   * @description It helps the user to fetch all of the created articles.
+   * @param  {object} req - The request object
+   * @param  {object} res - The response object
+   * @returns {object} It returns the request's response object
+   */
+
+  static async findAll(req, res, next) {
+    const { user: { id = null } = {} } = req;
+    const { author, favorited, tag } = req.query;
+    const { page = 1, limit = 20 } = req.query;
+    try {
+      const all = await Article.findAll({
+        offset: (Number(page) - 1) * Number(limit),
+        limit,
+        order: [["createdAt", "DESC"]],
+        where: { deletedAt: null },
+        attributes: { exclude: ["deletedAt"] },
+        include: [
+          {
+            model: User,
+            as: "author",
+            attributes: ["username", "bio", "image"],
+            where: author ? { username: author } : {},
+            include: [
+              { model: User, as: "followers", attributes: ["id", "username"] }
+            ]
+          },
+          { model: User, as: "likes", attributes: ["username"], through: {} },
+          { model: Tag, attributes: ["name"], as: "tagsList", through: {} }
+        ]
+      });
+      if (all.length) {
+        const articles = all.map(each => {
+          const { followers, username, bio, image } = each.get().author;
+          const { tagsList, likes, ...article } = each.get();
+          const following = !!_.dropWhile(followers, f => f.id !== id).length;
+          return {
+            ...article,
+            likes: likes.map(like => like.username),
+            author: { following, username, bio, image },
+            favorited: !!likes.length,
+            favoritesCount: likes.length,
+            tagsList: tagsList.map(t => t.get().name || null)
+          };
+        });
+        const articlesCount = articles.length;
+        return author || favorited || tag
+          ? next(articles)
```

Comment:
@Musigwa Can't we handle the filter in the same callback instead of triggering a second one?

Reply:
Can you please clarify that with some technical reasons why we should put everything in the same request handler? Because I have put it aside because the `findAll` handler only should fetch all of the data and pass control to the next handlers to filter them.

Doing this includes the SRP (of OOP concepts), and leaves the code clean that adds value to maintainability of the codebase.

Reply:
@Musigwa It was somehow confusing. I got your point